Link errors building debugmozjs magicleap

Lots of missing stl symbols from linking against a debugmozjs build for magicleap, e.g.
```
...
/Users/ajeffrey/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1582: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::size() const'
/Users/ajeffrey/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1582: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::max_size() const'
/Users/ajeffrey/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1590: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::setstate(unsigned int)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Oh joy.

I ran into this; here's the full log:
<details>

```
godot:servo jdm$ MLCERT=~/Downloads/servotestingapp.cert  MAGICLEAP_SDK=~/MagicLeap/mlsdk/v0.17.0/ ./mach package --dev --magicleap
[Servo2D] Compiling main.cpp...
[Servo2D] Compiling Servo2D.cpp...
[Servo2D] Compiling scenesGen.cpp...
[Servo2D] Compiling SceneDescriptor.cpp...
[Servo2D] Linking program Servo2D...
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `ComputeIsJITBroken()':
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1479: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::basic_string()'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1485: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::c_str() const'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1487: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::find(char const*, unsigned long) const'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1494: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::operator=(char const*)'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1498: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::find(char const*, unsigned long) const'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1509: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::find(char const*, unsigned long) const'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1519: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::fail() const'
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.61.6/mozjs/js/src/vm/JSContext.cpp:1519: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::eof() const'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `basic_ifstream':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:1019: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::basic_ios()'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `basic_istream':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:191: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::init(std::__1::basic_streambuf<char, std::__1::char_traits<char> >*)'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `basic_ifstream':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:1062: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::setstate(unsigned int)'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `std::__1::basic_istream<char, std::__1::char_traits<char> >& std::__1::getline<char, std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_istream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&)':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1608: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::widen(char) const'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `std::__1::basic_filebuf<char, std::__1::char_traits<char> >::sync()':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:860: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:860: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:883: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:883: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:889: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:889: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:892: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:892: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:896: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:896: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `std::__1::basic_filebuf<char, std::__1::char_traits<char> >::underflow()':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:590: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:592: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:592: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:594: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:594: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:596: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:596: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:599: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::egptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:599: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:600: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:603: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:604: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:605: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:606: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:629: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:630: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:635: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:637: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:639: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:639: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:640: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:646: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:647: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `std::__1::basic_filebuf<char, std::__1::char_traits<char> >::pbackfail(int)':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:656: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::eback() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:656: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:664: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:667: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::gptr() const'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `std::__1::basic_filebuf<char, std::__1::char_traits<char> >::overflow(int)':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:682: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:683: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::epptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:686: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:688: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:691: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:691: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:695: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:695: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:696: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:709: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:709: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:711: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:715: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:715: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:716: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:726: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:727: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::epptr() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:727: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::pbase() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/fstream:727: undefined reference to `std::__1::basic_streambuf<char, std::__1::char_traits<char> >::__pbump(long)'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(Unified_cpp_js_src35.o): In function `std::__1::basic_istream<char, std::__1::char_traits<char> >& std::__1::getline<char, std::__1::char_traits<char>, std::__1::allocator<char> >(std::__1::basic_istream<char, std::__1::char_traits<char> >&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&, char)':
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1564: undefined reference to `std::__1::basic_istream<char, std::__1::char_traits<char> >::sentry::operator bool() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1566: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::clear()'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1571: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::rdbuf() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1582: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::size() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1582: undefined reference to `std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >::max_size() const'
/Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/include/istream:1590: undefined reference to `std::__1::basic_ios<char, std::__1::char_traits<char> >::setstate(unsigned int)'
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(RemoveDynamicIndexing.o): In function `sh::TType const* sh::StaticType::Get<(sh::TBasicType)2, (sh::TPrecision)3, (sh::TQualifier)12, (unsigned char)1, (unsigned char)1>()':
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/StaticType.h:101:(.text._ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv[_ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv]+0x24): relocation truncated to fit: R_AARCH64_CALL26 against symbol `__cxa_guard_acquire' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/StaticType.h:102:(.text._ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv[_ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv]+0x88): relocation truncated to fit: R_AARCH64_CALL26 against symbol `memcpy@@LIBC' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/usr/lib/libc.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/StaticType.h:101:(.text._ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv[_ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv]+0x90): relocation truncated to fit: R_AARCH64_CALL26 against symbol `__cxa_guard_release' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/StaticType.h:101:(.text._ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv[_ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv]+0xc0): relocation truncated to fit: R_AARCH64_CALL26 against symbol `__cxa_guard_abort' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/StaticType.h:101:(.text._ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv[_ZN2sh10StaticType3GetILNS_10TBasicTypeE2ELNS_10TPrecisionE3ELNS_10TQualifierE12ELh1ELh1EEEPKNS_5TTypeEv]+0xc8): relocation truncated to fit: R_AARCH64_CALL26 against symbol `_Unwind_Resume' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(RemoveDynamicIndexing.o): In function `sh::RemoveDynamicIndexing(sh::TIntermNode*, sh::TSymbolTable*, sh::PerformanceDiagnostics*)':
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/RemoveDynamicIndexing.cpp:546:(.text._ZN2sh21RemoveDynamicIndexingEPNS_11TIntermNodeEPNS_12TSymbolTableEPNS_22PerformanceDiagnosticsE+0xd8): relocation truncated to fit: R_AARCH64_CALL26 against symbol `_Unwind_Resume' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(RemoveDynamicIndexing.o): In function `sh::(anonymous namespace)::RemoveDynamicIndexingTraverser::insertHelperDefinitions(sh::TIntermNode*)':
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/RemoveDynamicIndexing.cpp:313:(.text._ZN2sh12_GLOBAL__N_130RemoveDynamicIndexingTraverser23insertHelperDefinitionsEPNS_11TIntermNodeE+0xe4): relocation truncated to fit: R_AARCH64_CALL26 against symbol `std::__1::basic_ostream<char, std::__1::char_traits<char> >::operator<<(int)' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/RemoveDynamicIndexing.cpp:313:(.text._ZN2sh12_GLOBAL__N_130RemoveDynamicIndexingTraverser23insertHelperDefinitionsEPNS_11TIntermNodeE+0x15c): relocation truncated to fit: R_AARCH64_CALL26 against symbol `__assert2@@LIBC' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/usr/lib/libc.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/RemoveDynamicIndexing.cpp:313:(.text._ZN2sh12_GLOBAL__N_130RemoveDynamicIndexingTraverser23insertHelperDefinitionsEPNS_11TIntermNodeE+0x954): relocation truncated to fit: R_AARCH64_CALL26 against symbol `_Unwind_Resume' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/src/servo/support/magicleap/Servo2D/../../../target/magicleap/aarch64-linux-android/debug/libmlservo.a(RemoveDynamicIndexing.o): In function `sh::(anonymous namespace)::RemoveDynamicIndexingTraverser::visitBinary(sh::Visit, sh::TIntermBinary*)':
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/RemoveDynamicIndexing.cpp:371:(.text._ZN2sh12_GLOBAL__N_130RemoveDynamicIndexingTraverser11visitBinaryENS_5VisitEPNS_13TIntermBinaryE+0x110): relocation truncated to fit: R_AARCH64_CALL26 against symbol `std::__1::basic_ostream<char, std::__1::char_traits<char> >::operator<<(int)' defined in .text section in /Users/jdm/MagicLeap/mlsdk/v0.17.0/lumin/stl/libc++-lumin/lib/libc++.so
/Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/mozangle-0.1.7/gfx/angle/checkout/src/compiler/translator/RemoveDynamicIndexing.cpp:371:(.text._ZN2sh12_GLOBAL__N_130RemoveDynamicIndexingTraverser11visitBinaryENS_5VisitEPNS_13TIntermBinaryE+0x1b8): additional relocation overflows omitted from the output
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/jdm/src/servo/target/magicleap/aarch64-linux-android/debug/debug_lumin_clang-3.8_aarch64/Servo2D.mk:23: /Users/jdm/src/servo/target/magicleap/aarch64-linux-android/debug/debug_lumin_clang-3.8_aarch64/Servo2D] Error 1
```

</details>

--- a/components/script/script_runtime.rs
+++ b/components/script/script_runtime.rs
@@ -685,11 +685,11 @@ unsafe fn set_gc_zeal_options(cx: *mut JSContext) {
     use js::jsapi::{JS_SetGCZeal, JS_DEFAULT_ZEAL_FREQ};
 
     let level = match pref!(js.mem.gc.zeal.level) {
-        Some(level @ 0...14) => level as u8,
+        level @ 0...14 => level as u8,
         _ => return,
     };
     let frequency = match pref!(js.mem.gc.zeal.frequency) {
-        Some(frequency) if frequency >= 0 => frequency as u32,
+        frequency if frequency >= 0 => frequency as u32,
         _ => JS_DEFAULT_ZEAL_FREQ,
     };
     JS_SetGCZeal(cx, level, frequency);

--- a/python/servo/build_commands.py
+++ b/python/servo/build_commands.py
@@ -212,11 +212,6 @@ class MachCommands(CommandBase):
             print("Please specify either --target or --android.")
             sys.exit(1)
 
-        # https://github.com/servo/servo/issues/22069
-        if debug_mozjs and magicleap:
-            print("Please specify either --debug-mozjs or --magicleap.")
-            sys.exit(1)
-
         if android:
             target = self.config["android"]["target"]
 
@@ -576,6 +571,7 @@ class MachCommands(CommandBase):
             env.setdefault("HARFBUZZ_SYS_NO_PKG_CONFIG", "1")
             env.setdefault("PKG_CONFIG_ALLOW_CROSS", "1")
             env.setdefault("CMAKE_TOOLCHAIN_FILE", path.join(ml_support, "toolchain.cmake"))
+            env.setdefault("_LIBCPP_INLINE_VISIBILITY", "__attribute__((__always_inline__))")
 
             # The Open SSL configuration
             env.setdefault("OPENSSL_DIR", path.join(target_path, target, "native", "openssl"))

--- a/support/magicleap/Servo2D/Servo2D.mabu
+++ b/support/magicleap/Servo2D/Servo2D.mabu
@@ -14,10 +14,14 @@ LIBPATHS.debug = \
 LIBPATHS.release = \
     ../../../target/magicleap/aarch64-linux-android/release
 
+LDFLAGS.device = \
+    -L$(MLSDK)/lumin/stl/libc++/lib
+
 STLIBS = \
     mlservo
 
 SHLIBS = \
+    c++abi \
     log \
     z
 

--- a/support/magicleap/fake-ld.sh
+++ b/support/magicleap/fake-ld.sh
@@ -11,7 +11,7 @@ set -o pipefail
 MAGICLEAP_TOOLCHAIN=${MAGICLEAP_TOOLCHAIN:-"${MAGICLEAP_SDK}/tools/toolchains"}
 TARGET=${TARGET:-"aarch64-linux-android"}
 LD=${LD:-"${MAGICLEAP_TOOLCHAIN}/bin/${TARGET}-ld"}
-LDFLAGS=${LDFLAGS:-"-L${MAGICLEAP_SDK}/lumin/stl/libc++/lib -L${MAGICLEAP_SDK}/lumin/usr/lib -L${MAGICLEAP_TOOLCHAIN}/lib/gcc/${TARGET}/4.9.x ${MAGICLEAP_SDK}/lumin/usr/lib/crtbegin_so.o"}
+LDFLAGS=${LDFLAGS:-"-L${MAGICLEAP_SDK}/lumin/stl/libc++/lib -L${MAGICLEAP_SDK}/lumin/usr/lib -L${MAGICLEAP_TOOLCHAIN}/lib/gcc/${TARGET}/4.9.x"}
 
 # Remove the -landroid flag, grr
 ARGS=("$@")

